src/UIComponents/SaveModel.jsx

```diff
@@ -2,23 +2,48 @@
 import PropTypes from "prop-types";
 import React, { Component } from "react";
 import { connect } from "react-redux";
+import { setTrainedModelDetails, getTrainedModelDataToSave } from "../redux";
 
 class SaveModel extends Component {
   static propTypes = {
     saveTrainedModel: PropTypes.func,
-    trainedModel: PropTypes.object
+    trainedModel: PropTypes.object,
+    setTrainedModelDetails: PropTypes.func,
+    trainedModelDetails: PropTypes.object,
+    dataToSave: PropTypes.object
+  };
+
+  handleChange = (event, field) => {
+    const trainedModelDetails = this.props.trainedModelDetails;
+    trainedModelDetails[field] = event.target.value;
+    this.props.setTrainedModelDetails(trainedModelDetails);
   };
 
   onClickSave = () => {
-    alert("Don't get too excited, this doesn't do anything yet :)");
-    this.props.saveTrainedModel();
+    this.props.saveTrainedModel(this.props.dataToSave);
+    alert(
+      "Ok, you can get a little excited now; this saves the trained model to an s3 bucket!"
+    );
   };
 
   render() {
     return (
       <div>
         <br />
         <br />
+        <label>
+          Model name:
+          <input
+            type="text"
+            onChange={event => this.handleChange(event, "name")}
```

Comment:
It might be worth making this a controlled component where the value is set via the state, as described at https://reactjs.org/docs/forms.html

Reply:
Why? Then we'd have to keep the component's internal state consistent with the Redux state coming in as a prop. Am I missing something?

Reply:
It is likely fine to leave this as-is for now.  

The challenge with uncontrolled components is that they don't follow the React model of having state flow in one direction, and instead there is some state inside the HTML component that is outside of the React/Redux state, and can in fact drift out of sync.  I've had issues with more complex third-party components that have their own internal state which we can't control.

Reply:
Ah, I see. I think I'll merge as is for the sake of getting something working to play with and note that we can update later.

Reply:
For what it's worth, https://github.com/code-dot-org/ml-playground/pull/88 is a good example of why controlled is useful.